examples/Attic & Button/app/controllers/demo_controller.rb

```diff
@@ -0,0 +1,266 @@
+# frozen_string_literal: true
+
+#
+#    Copyright 2018, Optimizely and contributors
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+class DemoController < ApplicationController
+  before_action :authenticate_user!, except: %i[new create logout guest_shop]
+  before_action :initialize_optimizely_client!, only: %i[shop buy cart payment checkout_payment]
+  before_action :session_exists?, only: [:guest_shop]
+  before_action :product_exists!, only: %i[buy update_cart]
+  before_action :discount_feature_enabled?, only: :cart
+
+  def create
+    error_response("Email or Password can't be blank!") unless params[:email].present? && params[:password].present?
+    error_response('Invalid Email!') unless params[:email] =~ /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z]+)*\.[a-z]+\z/i
+    error_response("Experiment key can't be blank! add in optimizely_config.yml!") unless OPTIMIZELY_CONFIG['experiment_key'].present?
+    session[:current_user] = {
+      email: params[:email],
+      user_id: SecureRandom.hex(10),
+      name: params[:email].split('@').first,
+      domain: params[:email].split('@').last,
+      cart: []
+    }
+  end
+
+  def guest_shop
+    unless session[:current_user] && session[:current_user]['user_id']
+      delete_session!
+      session[:current_user] = {
+        user_id: SecureRandom.hex(10),
+        cart: []
+      }.as_json
+    end
+    redirect_to shop_path(user_id: session[:current_user]['user_id'])
+  end
+
+  def shop
+    @variation_key = session[:variation_key]
+    unless @variation_key
+      begin
+        @variation_key, succeeded = @optimizely_service.activate_service!(
+          @current_user,
+          OPTIMIZELY_CONFIG['experiment_key']
+        )
+        if succeeded
+          if @variation_key
+            session[:variation_key] = @variation_key
+          else
+            flash.now[:error] = "Failed to create variation using Experiment key: #{OPTIMIZELY_CONFIG['experiment_key']}!"
+          end
+        else
+          flash[:error] = @optimizely_service.errors
+        end
+      rescue StandardError => error
+        flash.now[:error] = "Failed to load datafile using Project ID: #{OPTIMIZELY_CONFIG['project_id']} (#{error})!"
+      end
+    end
+    @products = if @variation_key == 'sort_by_price'
```

Comment:
Make 'sort_by_price' a constant in config yml file

Reply:
Done